sympy/matrices/reductions.py

```diff
@@ -245,6 +247,35 @@ def complexity(i):
     return len(pivots)
 
 
+def _check_expr(M):
```

Comment:
@oscarbenjamin, will this helper work?
Thanks.

Reply:
Maybe.

Reply:
Would it be better to add the helper in `common.py` instead?
Thanks.

Reply:
Perhaps it would be better as a method on Matrix. I'm not sure which particular class it should be added to.